solr/core/src/java/org/apache/solr/cloud/ZkController.java

```diff
@@ -407,7 +408,11 @@ public ZkController(
 
       // note: Can't read cluster properties until createClusterState ^ is called
       final String urlSchemeFromClusterProp =
-          zkStateReader.getClusterProperty(ZkStateReader.URL_SCHEME, ZkStateReader.HTTP);
+          zkStateReader.getClusterProperty(
```

Comment:
When SSL is enabled we set this java sys prop `"solr.jetty.https.port"`(ie HTTPS_PORT_PROP)